### ISSUE TITLE
Inconsistent behaviour with Namespaces

### ISSUE THREAD

**Issue by alejandro-perez** [xmldom/xmldom#19]
I've detected what I believe is an inconsistent behaviour when generating/parsing documents with different namespaces.

How to reproduce:
```
var DOMParser = require('xmldom').DOMParser;
var doc = new DOMParser().parseFromString('<xml><A></A></xml>');
var a = doc.getElementsByTagName('A')[0];
var b = doc.createElementNS('test1', 'b');
var c = doc.createElementNS('test2', 'c');
var d = doc.createElementNS('test2', 'd');
b.appendChild(c);
b.appendChild(d);
a.appendChild(b);
doc.toString();
'<xml><A><b xmlns="test1"><c xmlns="test2"/><d/></b></A></xml>'
```
Note how element `<d>` (which we explicitly indicated should have namespace 'test2') has no namespace.
Now, if we just parse this document and print it, we have that:
```
var doc2 = new DOMParser().parseFromString(doc.toString());
doc2.toString()
'<xml><A><b xmlns="test1"><c xmlns="test2"/><d xmlns="test1"/></b></A></xml>'
```
In this second document, which should be identical to the first one, element `<d>` has namespace 'test1'.

I believe the error is making `<d>` lose the namespace definition, when it won't inherit the one from `<c>` as they are at the same level.

**Comment by alejandro-perez** [comment]
This is what Chromium generates using a similar sequence of commands:
```
"<xml><A><b xmlns="test1"><c xmlns="test2"/><d xmlns="test2"/></b></A></xml>"
```

**Comment by karfau** [comment]
This might be related or at least similar to what is described (less obvious) in https://github.com/jindw/xmldom/issues/215 which has a PR attached/linked.

**Comment by karfau** [comment]
I just verified that this issue was resolved by #268.
